### PR TITLE
Fix: Use dbcontext factory for migration bundle

### PR DESCRIPTION
**Pull request by wojtek2288**
Previous configuration would require changes in `app_config` terraform common module due to migrations access policy being required in `AddAppConfigurationFromAzureKeyVaultOnNonDevelopmentEnvironment` which was ran by ef previously.

### CODE REVIEW

File: backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs
Comment (wojtek2288) [inline backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs:1]: We might want to move this to corelib

File: backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        return dataSourceBuilder.Build();
+    }
+
+    public ExamplesDbContext CreateDbContext(string[] args)
+    {
+        var optionsBuilder = new DbContextOptionsBuilder<ExamplesDbContext>();
+
+        optionsBuilder.UseNpgsql(DataSource.CreateConnection());
```
Comment (Saancreed) [inline backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs:35]: 1. I believe you can pass the data source directly.
2. We should probably add `SetPostgresVersion(15, 0)` here.
3. Can we somehow make use of `args` here? Does `dotnet ef` pass anything interesting here on its own that we could use as primary source of connection string with fallback to env only if it's missing?

**Review by Saancreed — COMMENTED** [review]
:heart: but see the comment.

File: backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs (hunk 1 above)
Comment (wojtek2288) [reply backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs:35]: 3. not really it's empty array when running `dotnet ef migrations bundle` I guess it for in code usage.

Comment (Saancreed) [reply backend/src/Examples/ExampleApp.Examples.Services/DataAccess/ExamplesDbContextFactory.cs]: Not always, you can pass some yourself by calling it like so: `dotnet ef migrations bundle --force -- --myarg value`, then we will get `["--myarg", "value"]` in `args` but I was hoping to find something interesting there by default; alas, no luck there.

**Review by Saancreed — APPROVED** [review]

**Pull request merged**